### PR TITLE
151 - Fix issues with wonky ID input

### PR DESCRIPTION
**Pull request by rodalfus**
### Issue

**Steps to reproduce**:
(1):

Open login screen
copy your ID (e.g. ga12ab)
paste the ID in the first textfield
You will see that the whole ID is pasted into one textfield.

(2):

Open login screen
Input the first two characters of your ID and then select the first textfield again
Input another character
You will see that it is possible to input more than the expected two characters into the textfield.

**How this fixes the issue**:

- Specify number of characters fitting into each textfield explicitly
- Use the replacement string's indexes to find out where to put a given character

This fixes the following issue(s): 
#151

### CODE REVIEW

File: TUM Campus App/LoginViewController.swift [hunk 1]
```diff
@@ -58,39 +58,53 @@ extension LoginViewController: UITextFieldDelegate, TokenFetcherControllerDelega
...
+        if textFieldContentsAreValid() {
+            self.view.endEditing(true)
+            performSegue(withIdentifier: "waitForConfirmation", sender: self)
+        }
+        
+        return false
+    }
+    
+    private func textFieldContentsAreValid() -> Bool {
+        guard let firstText = firstTextField.text, let numbers = numbersTextField.text, let secondText = secondTextField.text, let _ = Int(numbers)
```
Comment (mathiasquintero) [inline TUM Campus App/LoginViewController.swift]: Please have these in separate lines. Just a new line after each comma and then the else block also in a new line

File: TUM Campus App/LoginViewController.swift [hunk 2]
```diff
@@ -58,39 +58,53 @@ extension LoginViewController: UITextFieldDelegate, TokenFetcherControllerDelega
...
+            return false
+        } else if string.characters.count < range.length {
+            return true
+        }
+    
+        var mutableReplaced = replaced
+        
+        for i in textFieldIndex..<textFields.endIndex {
+            let startIndex = mutableReplaced.startIndex
+            let endIndex = replaced.index(startIndex, offsetBy: min(mutableReplaced.characters.count, numberOfCharacterPerTextField[i]) - 1)
```
Comment (mathiasquintero) [inline TUM Campus App/LoginViewController.swift]: put each argument in a new line. As a rule of thumb. If I can't read it without scrolling maybe a new line would help

File: TUM Campus App/LoginViewController.swift [hunk 3]
```diff
@@ -58,39 +58,53 @@ extension LoginViewController: UITextFieldDelegate, TokenFetcherControllerDelega
...
+            let range = ClosedRange(uncheckedBounds: (lower: startIndex, upper: endIndex))
```
Comment (mathiasquintero) [inline TUM Campus App/LoginViewController.swift]: Uhhh! Unchecked Bounds, scary stuff... ;) Are you sure about this. I understand that you don't want the range to throw a fatalError if endIndex < startIndex.

Care to elaborate on why that happens?

Comment (rodalfus) [reply TUM Campus App/LoginViewController.swift]: Hmm, what would be the alternative to using `uncheckedBounds`? The `endIndex < startIndex` would only ever be true if one of the `numberOfCharactersPerTextField` were to be `0` or less. Perhaps I should be explicit about that being an array of unsigned integers.

Comment (mathiasquintero) [reply TUM Campus App/LoginViewController.swift]: `(startIndex..<endIndex)` should also work, but this is fine

**Pull request merged**